store/copr/batch_coprocessor.go

```diff
@@ -102,46 +103,82 @@ func (rs *batchCopResponse) RespTime() time.Duration {
 // 2. for the remaining regions:
 //    if there is only 1 available store, then put the region to the related store
 //    otherwise, use a greedy algorithm to put it into the store with highest weight
-func balanceBatchCopTask(originalTasks []*batchCopTask) []*batchCopTask {
+func balanceBatchCopTask(ctx context.Context, kvStore *kvStore, originalTasks []*batchCopTask, isMPP bool) []*batchCopTask {
 	if len(originalTasks) <= 1 {
 		return originalTasks
 	}
+	cache := kvStore.GetRegionCache()
 	storeTaskMap := make(map[uint64]*batchCopTask)
+	// storeCandidateRegionMap stores all the possible store->region map. Its content is
+	// store id -> region signature -> region info. We can see it as store id -> region lists.
 	storeCandidateRegionMap := make(map[uint64]map[string]RegionInfo)
 	totalRegionCandidateNum := 0
 	totalRemainingRegionNum := 0
 
-	for _, task := range originalTasks {
-		taskStoreID := task.regionInfos[0].AllStores[0]
-		batchTask := &batchCopTask{
-			storeAddr:   task.storeAddr,
-			cmdType:     task.cmdType,
-			ctx:         task.ctx,
-			regionInfos: []RegionInfo{task.regionInfos[0]},
+	if !isMPP {
+		for _, task := range originalTasks {
+			taskStoreID := task.regionInfos[0].AllStores[0]
+			batchTask := &batchCopTask{
+				storeAddr:   task.storeAddr,
+				cmdType:     task.cmdType,
+				ctx:         task.ctx,
+				regionInfos: []RegionInfo{task.regionInfos[0]},
+			}
+			storeTaskMap[taskStoreID] = batchTask
+		}
+	} else {
+
+		// decide the available stores
+		stores := cache.RegionCache.GetTiFlashStores()
+
+		for _, s := range stores {
+			aliveReq := tikvrpc.NewRequest(tikvrpc.CmdMPPAlive, &mpp.IsAliveRequest{}, kvrpcpb.Context{})
```

Comment:
could tiflash use grpc's health check like  https://github.com/tikv/client-go/blob/2c7970b2b7e8ba70ecc58f3eaa1c33b7da68bea4/internal/locate/region_cache.go#L2249  instead of use new command~?

if so maybe we can set up unified health check mechanism in future